packages/hardhat-core/src/internal/hardhat-network/provider/vm/dual.ts

```diff
@@ -224,3 +327,39 @@ function assertEqualRunTxResults(
     }
   }
 }
+
+function assertEqualAccounts(
+  address: Address,
+  ethereumJSAccount: Account,
+  rethnetAccount: Account
+) {
+  if (ethereumJSAccount.balance !== rethnetAccount.balance) {
+    console.trace(`Account: ${address}`);
+    console.trace(
+      `Different balance: ${ethereumJSAccount.balance} !== ${rethnetAccount.balance}`
+    );
+    throw new Error("Different balance");
+  }
+
+  if (!ethereumJSAccount.codeHash.equals(rethnetAccount.codeHash)) {
+    console.trace(
+      `Different codeHash: ${ethereumJSAccount.codeHash} !== ${rethnetAccount.codeHash}`
+    );
+    throw new Error("Different codeHash");
+  }
+
+  if (ethereumJSAccount.nonce !== rethnetAccount.nonce) {
+    console.trace(
+      `Different nonce: ${ethereumJSAccount.nonce} !== ${rethnetAccount.nonce}`
+    );
+    throw new Error("Different nonce");
+  }
+
+  // TODO: Add storageRoot to Rethnet
```

Comment:
Good call. The question I guess is: Do we want to always know the storage root when we query account data? It's extra computation (for now at least)

Reply:
Hmm, apparently we are not using it anywhere, so... maybe we just don't do this for now? I mean, let's not even open an issue. If it comes up as a necessity later, we can re-evaluate.

Reply:
Or, alternative idea: let's do it just because it lets us compare more things, but we drop it later when we remove ethereumjs?

I'm fine with either option, because the extra confidence this would give us might not be worth it. Up to you!

Reply:
I can add it easily. Already have a function for calculating the storage root. I proposed making it a separate API so that we can keep it, because people might want to manually retrieve a storage root for an account? 🤷‍♂️

Reply:
Sure, that seems sensible

Reply:
Resolved by https://github.com/NomicFoundation/hardhat/pull/3454